cpp/src/neighbors/detail/cagra/search_plan.cuh

```diff
@@ -193,10 +199,16 @@ struct search_plan_impl : public search_plan_impl_base {
     uint32_t _max_iterations = max_iterations;
     if (max_iterations == 0) {
       if (algo == search_algo::MULTI_CTA) {
-        _max_iterations = 1 + std::min(32 * 1.1, 32 + 10.0);  // TODO(anaruse)
+        constexpr uint32_t mc_itopk_size   = 32;
+        constexpr uint32_t mc_search_width = 1;
+        _max_iterations                    = mc_itopk_size / mc_search_width;
       } else {
-        _max_iterations =
-          1 + std::min((itopk_size / search_width) * 1.1, (itopk_size / search_width) + 10.0);
+        _max_iterations = itopk_size / search_width;
+      }
+      int64_t num_reachable_nodes = 1;
+      while (num_reachable_nodes < dataset_size) {
```

Comment:
I like the updated heuristics here, it seems to be a reasonable choice to relate the number of iterations to the dataset size.

Reply:
I think the appropriate number of search iterations depends on the size of dataset and the degree of the graph. The previous method of determining the number of iterations did not take these parameters into account, so I have made this change.